contracts/adapters/FixedRebalanceExtension.sol

```diff
@@ -110,23 +112,24 @@ contract FixedRebalanceExtension is BaseExtension {
      * @param _maturities                Relative maturities (i.e. "3 months") in seconds
      * @param _allocations               Relative allocations (i.e. 0.9 = 90%) with 18 decimals corresponding to the respective maturity
      */
-    function setAllocations(uint256[] memory _maturities, uint256[] memory _allocations) external onlyOperator {
-        _setAllocations(_maturities, _allocations);
+    function setAllocations(uint256[] memory _maturities, uint256[] memory _allocations, uint256[] memory _minPositions) external onlyOperator {
+        _setAllocations(_maturities, _allocations, _minPositions);
     }
 
     /**
      * ONLY ALLOWED CALLER: Rebalances the positions towards the configured allocation percentages.
      *
-     * @param _share                Relative share of the necessary trade volume to execute (allows for splitting the rebalance over multiple transactions
-     * @param _minPositions         Minimum positions (in set token units) for each maturity after rebalance. (slippage protection)
+     * @param _share                 Relative share of the necessary trade volume to execute (allows for splitting the rebalance over multiple transactions
+     * @param _rebalanceMinPositions Minimum positions (in set token units) for each maturity after this rebalance operation. 
+     * @dev Will revert if _rebalanceMinPositions is lower than the minPositions configured by the operator (weighted by share)
      */
-    function rebalance(uint256 _share, uint256[] memory _minPositions) external onlyAllowedCaller(msg.sender) {
+    function rebalance(uint256 _share, uint256[] memory _rebalanceMinPositions) external onlyAllowedCaller(msg.sender) {
```

Comment:
For _rebalanceMinPositions, are we able to derive that from existing storage so that we don't need the caller to pass it in? Ideally, if we were to open rebalance up to being called by anyone, then it shouldn't have any parameters. I guess if we could set _share and _rebalanceMinPositions, as storage variables, which can only be set by operator, then we'd can open rebalance up to being called by anyone? Such as chainlink keepers?

Reply:
Yes we can derive the minimum from a storage variable and we already do that. (see here:
https://github.com/IndexCoop/index-coop-smart-contracts/pull/125/files#diff-4e84502ce034d717a90159b1ab768a5f6eeafcb03e351853b5b1064675789211R239)

The idea here is that we use that storage variable as protection against malicious `rebalance` calls but then let the caller "overwrite" these values with higher ones to get the optimal slippage protection based on market conditions. 

Also in this current way we are quite flexible, for example if we set the `minPositions` to all zeroes then we basically keep the existing logic of letting the caller set arbitrary minPositions. (of course in that case we can't open it up to public calls).


I'm not sure if this might be a bit of over-engineering. But I thought if we just let the operator set these values then we have the tradeoff between either setting these values lower and paying a lot in slippage / sandwhich mev or setting them quite high but then having to continously update them to account for changing market conditions.


Does that make sense ? What do you think ?  @FlattestWhite

Reply:
I also tried to summarize this idea with different wording in the description / summary comment of the PR.